### PR TITLE
Eliminate android test log spam

### PR DESCRIPTION
**Pull request by reidbaker**
Set gradle to treat warnings as errors and suppress or fix all warnings in engine android tests. 

Fixes flutter/flutter#133070

After doing this work I was disappointed to realized that the only lint turned on was deprecration but this is still a step in the right direction. 

- Remove usages of deprecated junit.framework and replace with org.junit in KeyboardManagerTest, PlatformViewsControllerTest and SinglePresentationViewTest
- Annotate deprecated usages of Registrar
- Suppress warnings for getSystemWindowInsets and ensure roboletric config consistant with targetApi lint annotation, bump itDescribesTextFieldsWithTextAndHint to 28 because test actually fails on api 26
- Suppress warnings for DartExecutor.send
- Suppress warnings for ClipboardManager.set/hasText
- Suppress warnings for getWindowSystemUiVisibility, setMessageHandler, DartExecutor.send, InputMethodSubtype, and proof of concept migration from Roboletric.setupActivity
- Suppress deprecation warnings, set minsdk on tests that were checking for sdk version
- Suppress deprecation warnings in SpellCheckPluginTest
- Suppress deprecation warnings in MouseCursorPluginTest, set minsdk config to match target api
- Stop calling RuntimeEnvrionment.application and insted a call ApplicationProvider.getApplicationContext() in SingleViewPresentationTest
- Start calling FlutterView(Context, FlutterSurfaceView)
- Suppress deprecation warnings in PlatformPluginTest, getText, Robolectric.setup/buildActivity, system ui flags
- Suppress deprecation warnings in PlayStoreDeferredComponentManagerTest
- Suppress deprecation warnings in KeyboardChannelTest
- Suppress deprecation warnings in SettingsChannelTest
- Suppress deprecation warnings in ApplicationInfoLoaderTest
- Stop calling RuntimeEnvrionment.application and insted a call ApplicationProvider.getApplicationContext() in FlutterEngineGroupCacheTest
- Suppress deprecation warnings in FlutterAndroidComponentTest
- Suppress deprecation warnings in FlutterFragmentTest, more robolectric setup activity calls
- Suppress deprecation warnings in FlutterActivityAndFragmentDelegateTest
- Shadow.directlyOn is incompatible with java 17+, refactor to use reflectors and direct following https://github.com/robolectric/robolectric/pull/6598/files as an example Unsure if SplashShadowResources is used anywhere
- Enable warnings as errors
- Formatting

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java [hunk 1]
```diff
@@ -66,6 +66,8 @@ public class FlutterActivityAndFragmentDelegateTest {
   private FlutterEngine mockFlutterEngine;
   private FlutterActivityAndFragmentDelegate.Host mockHost;
 
+  @SuppressWarnings("deprecation")
+  // Robolectric.setupActivity
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java:70]: Consider extracting a `Activity GetMockActivity()` or similar that returns `Robolectric.setupActivity(Activity.class)` so you aren't suppressing the entire method.

Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java:70]: There should probably be a TODO with an issue link about replacing all use of this deprecated method.

File: shell/platform/android/test/io/flutter/embedding/android/FlutterAndroidComponentTest.java [hunk 2]
```diff
@@ -207,6 +207,8 @@ public void normalLifecycleStepsDoNotTriggerADetachFromFlutterEngine() {
     verify(delegate, never()).detachFromFlutterEngine();
   }
 
+  @SuppressWarnings("deprecation")
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterAndroidComponentTest.java:210]: Same.

File: shell/platform/android/test/io/flutter/embedding/android/FlutterFragmentTest.java [hunk 3]
```diff
@@ -283,6 +283,8 @@ public void itReturnsExclusiveAppComponent() {
     assertEquals(fragment.getExclusiveAppComponent(), delegate);
   }
 
+  @SuppressWarnings("deprecation")
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterFragmentTest.java:286]: Same.

File: shell/platform/android/test/io/flutter/embedding/android/SplashShadowResources.java [hunk 4]
```diff
@@ -8,22 +10,34 @@
 import org.robolectric.annotation.Implementation;
 import org.robolectric.annotation.Implements;
 import org.robolectric.annotation.RealObject;
-import org.robolectric.shadow.api.Shadow;
 import org.robolectric.shadows.ShadowResources;
+import org.robolectric.util.reflector.Direct;
+import org.robolectric.util.reflector.ForType;
 
+@SuppressWarnings("deprecation")
+// getDrawableInt
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/embedding/android/SplashShadowResources.java:18]: Do you mean `getDrawable(int)`? There's no `getDrawableInt` in this code.

File: shell/platform/android/test/io/flutter/embedding/engine/systemchannels/KeyboardChannelTest.java [hunk 5]
```diff
@@ -32,6 +32,8 @@ private static BinaryMessenger.BinaryReply sendToBinaryMessageHandler(
     return reply;
   }
 
+  @SuppressWarnings("deprecation")
+  // setMessageHandler is deprecated.
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/embedding/engine/systemchannels/KeyboardChannelTest.java:36]: Is there an issue we can reference for this? If we have a test verifying that our own code is calling our own deprecated method, it seems like there's a problem somewhere.

File: shell/platform/android/test/io/flutter/embedding/engine/systemchannels/SettingsChannelTest.java [hunk 6]
```diff
@@ -23,6 +23,8 @@ public class SettingsChannelTest {
   @Test
   @TargetApi(33)
   @Config(sdk = 33)
+  @SuppressWarnings("deprecation")
+  // DartExecutor.send is deprecated.
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/embedding/engine/systemchannels/SettingsChannelTest.java:27]: Can we reference an issue for all of these deprecated send calls (here and in other files)?

File: shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java [hunk 7]
```diff
@@ -1860,6 +1879,9 @@ public void autofill_doesNotCrashAfterClearClientCall() {
         .updateEditingState(anyInt(), any(), anyInt(), anyInt(), anyInt(), anyInt());
   }
 
+  @Config(minSdk = Build.VERSION_CODES.O)
+  @SuppressWarnings("deprecation")
+  // Robolectric.setupActivity.
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java:1884]: Helper for all of these in this file, to consolidate the deprecation warning and the TODO?

File: shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java [hunk 8]
```diff
@@ -1441,12 +1447,15 @@ public void autofill_hintText() {
...
   @Test
   public void autofill_onProvideVirtualViewStructure() {
     if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
       return;
     }
-
+    // Migrate to ActivityScenario by following https://github.com/robolectric/robolectric/pull/4736
```
Comment (stuartmorgan) [inline shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java:1458]: This looks like it should be a `// TODO`

File: shell/platform/android/test_runner/build.gradle [hunk 9]
```diff
@@ -22,7 +22,7 @@ rootProject.buildDir = project.property("build_dir")
 // 434 (at the time of this comment) deprecations are fixed.
 gradle.projectsEvaluated {
   tasks.withType(JavaCompile) {
-      options.compilerArgs << "-Xlint:deprecation" << "-Xmaxwarns" << "434"
+      options.compilerArgs << "-Xlint:deprecation" << "-Werror" << "-Xmaxwarns" << "434"
```
Comment (stuartmorgan) [inline shell/platform/android/test_runner/build.gradle:25]: Do you want to include a TODO with an issue link for adding more lint options?

**Review by stuartmorgan — APPROVED** [review]
Various small suggestions, but LGTM as a good step toward having fewer warnings!

File: shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java (hunk 7 above)
Comment (reidbaker) [reply shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java]: Good idea. Migrated to test method for the class.

**Pull request merged**